### ISSUE TITLE
Incorporate Panel App Oz

### ISSUE THREAD

**Issue by sksmi** [SACGF/variantgrid#82]
**Is your feature request related to a problem? Please describe.**
Australian diagnostic services are starting to switch to using Panel App Oz, but the current version of Panel App in VG is the UK version.

**Describe the solution you'd like**
Can we integrate Oz Panel App into VG (possibly all environs?)

**Describe alternatives you've considered**
Manually import panels. Happy to do this as a temp solution, if the above is time-consuming & will do for clinical V2.0 unless a patch is feasible. 

**Additional context**
Oz API docs here: https://panelapp.agha.umccr.org/#!API

**Comment by davmlaw** [comment]
Bump - Sarah wants for training docs

**Comment by davmlaw** [comment]
Didn't get it done in time to make upgrade (barely tested), but here's what I did so far - this shows "Arthrogryposis" which is in both:

http://144.6.229.160/genes/gene_grid/gene-list-panel-app-467/gene-list-panel-app-583



Still TODO:

* Need to deal with evidence (currently only works w/England)

**Comment by davmlaw** [comment]
Also added to gene symbol page

FYI this is what it looks like, it will be enabled in Shariant test, but NOT in prod (as we've hidden gene annotations)

![panel_app_screenshot.png](https://images.zenhubusercontent.com/5bb45cdce135e46293defe80/88a1e7b2-8d81-4a4d-a7f1-65f09255fdb1)

If you want this to go to eventually appear in prod let me know

Historical fixes are in new issue: https://github.com/SACGF/variantgrid_sapath/issues/57

**Comment by sksmi** [comment]
**FAIL**

SAP Test: generated a 500 error: http://frgeneseq02.imvs.sa.gov.au:90/analysis/5253/
Think it was due to me trying to simultaneously apply a PanelApp list for Genomics England at the same time as Oz PanelApp (hadn't thought to try that out before) - might need to restrict that from happening. 
![image.png](https://images.zenhubusercontent.com/5bc5465086f3a678d497c0be/71ff753d-acbd-48a2-958e-39a4e4271b98)

VG test: Tried a simple single PanelApp load in VG.test also failed. http://144.6.229.160/analysis/1948/
![image.png](https://images.zenhubusercontent.com/5bc5465086f3a678d497c0be/df5af699-d9af-4551-bb22-d7cc2bb3c22a)

**Comment by sksmi** [comment]
Note that Oz autism panel loads fine in http://144.6.229.160/analysis/1949/
Something to do with create from blank vs create from open analysis?

**Comment by davmlaw** [comment]
Failed 4 times in a second, then another 4 times in a second 4 seconds afterwards:

Feb 22 21:10:50 sapathtest gunicorn[1377278]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:50 sapathtest gunicorn[1377278]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:50 sapathtest gunicorn[1377278]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:50 sapathtest gunicorn[1377278]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:54 sapathtest gunicorn[1377284]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:54 sapathtest gunicorn[1377284]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:54 sapathtest gunicorn[1377284]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:54 sapathtest gunicorn[1377284]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.


```
File "/data/variantgrid/analysis/views/nodes/node_view.py", line 57, in form_valid
  self.object = form.save(commit=False)
File "/data/variantgrid/analysis/forms/forms_nodes.py", line 378, in save
  panel_app_panel_local_cache_gene_list = get_local_cache_gene_list(pap)
File "/data/variantgrid/genes/panel_app.py", line 155, in get_local_cache_gene_list
  pap_gene_list = PanelAppPanelLocalCacheGeneList.objects.create(panel_app_panel=panel_app_panel,
File "/usr/local/lib/python3.8/dist-packages/django/db/models/manager.py", line 85, in manager_method
  return getattr(self.get_queryset(), name)(*args, **kwargs)
File "/usr/local/lib/python3.8/dist-packages/django/db/models/query.py", line 447, in create
  obj.save(force_insert=True, using=self.db)
File "/usr/local/lib/python3.8/dist-packages/django_extensions/db/models.py", line 22, in save
  super().save(**kwargs)
File "/usr/local/lib/python3.8/dist-packages/django/db/models/base.py", line 750, in save
  self.save_base(using=using, force_insert=force_insert,
File "/usr/local/lib/python3.8/dist-packages/django/db/models/base.py", line 787, in save_base
  updated = self._save_table(
File "/usr/local/lib/python3.8/dist-packages/django/db/models/base.py", line 892, in _save_table
  results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
File "/usr/local/lib/python3.8/dist-packages/django/db/models/base.py", line 930, in _do_insert
  return manager._insert(
File "/usr/local/lib/python3.8/dist-packages/django/db/models/manager.py", line 85, in manager_method
  return getattr(self.get_queryset(), name)(*args, **kwargs)
File "/usr/local/lib/python3.8/dist-packages/django/db/models/query.py", line 1249, in _insert
  return query.get_compiler(using=using).execute_sql(returning_fields)
File "/usr/local/lib/python3.8/dist-packages/django/db/models/sql/compiler.py", line 1395, in execute_sql
  cursor.execute(sql, params)
File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 66, in execute
  return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
  return executor(sql, params, many, context)
File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 84, in _execute
  return self.cursor.execute(sql, params)
File "/usr/local/lib/python3.8/dist-packages/django/db/utils.py", line 90, in __exit__
  raise dj_exc_value.with_traceback(traceback) from exc_value
File "/usr/local/lib/python3.8/dist-packages/django/db/backends/utils.py", line 84, in _execute
  return self.cursor.execute(sql, params)
Feb 22 21:10:50 sapathtest gunicorn[1377278]: django.db.utils.IntegrityError: duplicate key value violates unique constraint "genes_panelapppanellocal_panel_app_panel_id_versi_2e10e775_uniq"
Feb 22 21:10:50 sapathtest gunicorn[1377278]: DETAIL:  Key (panel_app_panel_id, version)=(426, 3.963) already exists.
Feb 22 21:10:50 sapathtest gunicorn[1377278]: ERROR Internal Server Error: /analysis/node/view/0/41968/1/default/
```

**Comment by davmlaw** [comment]
Couldn't recreate normally, but looking at the time it occured (4 times within a second) was able to recreate by rapidly clicking the save button multiple times while it was loading the 1st time (it always takes much longer the 1st time as it has to contact PanelApp)

**Comment by davmlaw** [comment]
Fixed by implementing #190 - which is to save the form quickly, and then retrieve the gene lists via API when the node loads

Bless Sarah and her manic clicking - I think this was a good thing to do, having the UI freeze for 2 seconds per panel app panel (imagine if there were 5!) was bad
